spec/unit/queue_spec.rb

```diff
@@ -42,6 +42,15 @@ def new(options = {})
       expect(received.values.first).to eq sent
     end
 
+    it "gets the message with 'DelaySeconds' option" do
+      sent = send_message({ "DelaySeconds" => 1 })
+      received = receive_message
+      expect(received.values.first).to be_nil
+      sleep 1
```

Comment:
I understand that the sleep is necessary to test the DelaySeconds but it would be nicer if you stub where the original service is getting its current time from. That way the unit test does not have to wait around for a whole second, and you could test other time ranges if required.